File: android-exercises/6 - recycler_view/RecyclerView_java/app/src/main/java/com/bugrahankaramollaoglu/recyclerview_java/MainActivity.java
package com.bugrahankaramollaoglu.recyclerview_java;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.bugrahankaramollaoglu.recyclerview_java.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        String[] colors = {"purple", "pink", "green", "red","yellow","cyan","crimson","black"};

        ColorAdapter colorAdapter = new ColorAdapter(colors);
        binding.recyclerView.setAdapter(colorAdapter);
        binding.recyclerView.setLayoutManager(new LinearLayoutManager((this)));

    }
}

File: android-exercises/6 - recycler_view/RecyclerView_kotlin/app/src/main/java/com/bugrahankaramollaoglu/recyclerview_kotlin/MainActivity.kt
package com.bugrahankaramollaoglu.recyclerview_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.bugrahankaramollaoglu.recyclerview_kotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        val colorList = ArrayList<String>()
        colorList.add("purple")
        colorList.add("blue")
        colorList.add("yellow")
        colorList.add("green")
        colorList.add("red")

        val colorAdapter = ColorAdapter(colorList)
        binding.recycleView.layoutManager = LinearLayoutManager(this)
        binding.recycleView.adapter = colorAdapter
    }
}